assets/shaders/rally_point.wgsl

```diff
@@ -0,0 +1,27 @@
+#import bevy_pbr::mesh_view_bindings
+#import bevy_pbr::mesh_bindings
+
+struct CustomMaterial {
+    color: vec4<f32>,
+    pointiness: f32,
+    speed: f32,
+    length: f32,
+    spacing: f32,
+};
+
+@group(1) @binding(0)
+var<uniform> material: CustomMaterial;
+
+@fragment
+fn fragment(
+    #import bevy_pbr::mesh_vertex_output
+) -> @location(0) vec4<f32> {
+    let scaled_x: f32 = uv.x * length(mesh.model[0].xyz);
+    let offset_y: f32 = abs(uv.y - 0.5) * material.pointiness;
+    let scaled_time: f32 = globals.time * material.speed;
+    let total_length = material.length + material.spacing;
+    let value = scaled_x + offset_y - scaled_time;
```

Comment:
I would make it transparent if `(scaled_x + offset_y + material.spacing) > length(mesh.model[0].xyz)` so that the arrows disappear once the tip reaches the end (instead of it being clamped).